### DISCUSSION TITLE
Want to add onClick iframe option

### DISCUSSION THREAD

**Discussion by mzbu** [facebook/docusaurus#8400] [Q&A]
Hello,

Recently, I swizzled the footer to add a static button in the bottom right corner of the webpage that, when clicked, opens a new tab with a feedback form we created through HubSpot. This has worked great, but we've been looking to embed the form on Docusaurus so that when the button is clicked, an iframe with the form appears.  It currently looks something like this (the src is replaced for privacy):

#### index.js (button for the swizzled footer)
```
import React from 'react';
import ContactSupportIcon from '@mui/icons-material/ContactSupport';
 
export default function Feedback({ resource }) {
 return <div class="tooltip">
 <button id="btn1" class="hubspot-button">
 <div class="hubspot-icon" ><ContactSupportIcon sx={{ fontSize: 50 }} /></div>
 <span class="tooltiptext">Give us feedback</span>
</button>
<iframe id="iframe1"></iframe>
<script>
 const element = document.getElementById("btn1");
 element.addEventListener("click", myFunction);
 
 function myFunction(){
   document.getElementById('iframe1').innerHTML = <iframe> src="myhubspotlink" </iframe>
 }
</script>
</div>
```

I've tried a number of ways to accomplish this, but without much luck. Any help or guidance would be exceptional, thanks!

**Comment by mzbu** [comment]
I have also tried this:
```
import React from 'react';
import ContactSupportIcon from '@mui/icons-material/ContactSupport';

export default function Feedback({ resource }) {
  return <div class="tooltip">
  <button id="btn1" class="hubspot-button" onclick="myFunction()">
  <div class="hubspot-icon" ><ContactSupportIcon sx={{ fontSize: 50 }} /></div>
  <span class="tooltiptext">Give us feedback</span>
</button>
<div id="formdisplay"></div>
<script type="text/javascript">
  function myFunction(){
    document.getElementById("formdisplay").innerHTML = "<iframe src=\"myhubspotlink\" width=\"500\"></iframe>"
  };
</script>
</div>

}
```

but this throws this error, despite "formdisplay" being defined previously
![Screen Shot 2022-11-30 at 12 51 04 PM](https://user-images.githubusercontent.com/95442268/204886265-ed878283-f58d-47c7-a96a-a76454d92691.png)

**Comment by slorber** [comment]
Hi

Unfortunately it will be hard to help you because I'm not sure what you are trying to do. 

Make sure you understand enough of React to achieve your goal, it looks to me you are fighting against its principles. 
This video might help: https://www.youtube.com/watch?v=s-lumOeD2fk

most likely you don't need to add an extra script tag to achieve what you want, and should rather use React features like useState to decide when to display that iframe

You may get better support on Discord in case you are looking for casual React help
